### PR TITLE
Feature/pelagos 4165 add funding cycle dropdown

### PR DESCRIPTION
**Pull request by mickel1138**

### CODE REVIEW

File: assets/js/vue/components/FundingCyclesList.vue [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                this.researchGroupDisabled = true;
+                this.fundingCycles.forEach(fundingCycle => {
+                    if (fundingCycle.id == event.target.value) {
+                        this.researchGroups = fundingCycle.researchGroups;
+                        this.researchGroupDisabled = false;
+                    }
+                })
+            },
+            selectResearchGroup: function(event) {
+                if (event.target.value != "") {
```
Comment (praneethpr) [inline assets/js/vue/components/FundingCyclesList.vue:48]: We could just do this to check if value exists
```suggestion
                if (event.target.value) {
```

File: src/Controller/DefaultController.php [hunk 2]
```diff
@@ -30,14 +31,31 @@ class DefaultController extends AbstractController
...
+
+        foreach ($results as $fundingCycle) {
+            $data = array();
+            $data['id'] = $fundingCycle->getId();
+            $data['name'] = $fundingCycle->getName();
+
+            foreach ($fundingCycle->getResearchGroups() as $researchGroup) {
+                $rg['id'] = $researchGroup->getId();
+                $rg['name'] = $researchGroup->getName();
+                $data['researchGroups'][] = $rg;
```
Comment (praneethpr) [inline src/Controller/DefaultController.php:51]: We could use array_push

File: src/Controller/DefaultController.php [hunk 3]
```diff
@@ -30,14 +31,31 @@ class DefaultController extends AbstractController
...
-            $researchGroups = $this->get('doctrine')->getRepository(ResearchGroup::class)->findBy(array('id' => $fundingOrgFilter->getResearchGroupsIdArray() ));
+            $filter = array('fundingOrganization' => $fundingOrgFilter->getFilterIdArray());
+        }
+
+        $results = $this->get('doctrine')->getRepository(FundingCycle::class)->findBy($filter, array('name' => 'ASC'));
+
+        $fundingCycles = array();
+
+        foreach ($results as $fundingCycle) {
+            $data = array();
```
Comment (praneethpr) [inline src/Controller/DefaultController.php:44]: I think we could use some other variable name

File: src/Controller/DefaultController.php [hunk 4]
```diff
@@ -30,14 +31,31 @@ class DefaultController extends AbstractController
...
+            $data['id'] = $fundingCycle->getId();
+            $data['name'] = $fundingCycle->getName();
+
+            foreach ($fundingCycle->getResearchGroups() as $researchGroup) {
+                $rg['id'] = $researchGroup->getId();
```
Comment (praneethpr) [inline src/Controller/DefaultController.php:49]: Same here, expanded form of $rg would be good

File: assets/css/grp-home.css [hunk 5]
```diff
@@ -0,0 +1,19 @@
+.card-product {
```
[the remark this answers is not in the document]
Comment (mickel1138) [reply assets/css/grp-home.css:1]: gotta delete it yes, the file I mean.

File: assets/js/vue/components/FundingCyclesList.vue [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        },
+        data() {
+            return {
+                researchGroups: [],
+                researchGroupDisabled: true,
+            }
+        },
+        methods: {
+            populateResearchGroups: function(event) {
+                this.researchGroups = [];
```
[the remark this answers is not in the document]
Comment (mickel1138) [reply assets/js/vue/components/FundingCyclesList.vue:38]: I'm erasing it here. On purpose.

File: src/Controller/DefaultController.php (hunk 2 above)
Comment (praneethpr) [reply src/Controller/DefaultController.php:51]: Yes, It looks like using $array[] seems to be faster than using array_push(). Didn't know that

File: assets/js/vue/components/FundingCyclesList.vue [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        <select @change="populateResearchGroups" class="form-control">
+            <option value="" selected>[Please select a Funding Cycle]</option>
+            <option v-for="fundingCycle in fundingCycles" :value="fundingCycle.id" :key="fundingCycle.id">{{ fundingCycle.name }}</option>
+        </select>
+
+        <label class="col-form-label col-form-label-lg">
+            Research Groups
+        </label>
+
+        <select @change="selectResearchGroup" class="form-control" :disabled="researchGroupDisabled">
```
Comment (praneethpr) [inline assets/js/vue/components/FundingCyclesList.vue:16]: Doesn't work if the user wants the 1st research group in the list

**Pull request merged**